app/config/ApiContext.scala

```diff
@@ -20,14 +20,14 @@ import com.google.inject.{Inject, Singleton}
 import play.api.Configuration
 
 @Singleton
-class AppContext @Inject()(configuration: Configuration) {
+class ApiContext @Inject()(configuration: Configuration) {
   private val apiScopeConfigKey = "api.definition.scope"
   private val apiContextConfigKey = "api.context"
   private val apiWhitelistedServicesConfigKey = "api.access.whitelistedApplicationIds"
   private val apiAccessKey = "api.access"
   private def apiConfigException(apiConfigKey: String) = new IllegalStateException(s"$apiConfigKey is not configured")
-  lazy val apiScopeKey: String = configuration.getString(apiScopeConfigKey).getOrElse(throw apiConfigException(apiScopeConfigKey))
-  lazy val apiWhitelistKey: String = configuration.getString(apiWhitelistedServicesConfigKey).getOrElse(throw apiConfigException(apiWhitelistedServicesConfigKey))
-  lazy val apiContext: String = configuration.getString(apiContextConfigKey).getOrElse(throw apiConfigException(apiContextConfigKey))
-  lazy val apiAccess: Option[Configuration] = configuration.getConfig(apiAccessKey)
-}
\ No newline at end of file
+  lazy val apiScopeKey: String = configuration.getOptional[String](apiScopeConfigKey).getOrElse(throw apiConfigException(apiScopeConfigKey))
+  lazy val apiWhitelistKey: String = configuration.getOptional[String](apiWhitelistedServicesConfigKey).getOrElse(throw apiConfigException(apiWhitelistedServicesConfigKey))
+  lazy val apiContext: String = configuration.getOptional[String](apiContextConfigKey).getOrElse(throw apiConfigException(apiContextConfigKey))
+  lazy val apiAccess: Option[Configuration] = configuration.getOptional[Configuration](apiAccessKey)
```

Comment:
Although it returns an Option[Configuration], should we be doing a getOrElse also? Any issues returning nothing for a Configuration?

Reply:
inlined logic in to apiConfig which has removed the case class that required this.